clustering.Rmd

```diff
@@ -45,95 +49,147 @@ use the subgroups to generate new questions about the data and follow up with a
 
-penguin_data <- data |> select(flipper_length_scaled, bill_length_scaled)
+penguin_data <- data |> select(flipper_length_standardized, bill_length_standardized)
+write_csv(penguin_data, "data/penguins_standardized.csv")
 ```
 
 \index{seed!set.seed}
 
-```{r 10-clustering, warning = F, message=F}
+```{r 10-clustering, warning = FALSE, message = FALSE}
 library(tidyverse)
-
 set.seed(1)
+```
 
+Now we can load and preview the data.
+
+```{r, message = FALSE, warning = FALSE}
+penguin_data <- read_csv("data/penguins_standardized.csv")
 penguin_data
 ```
 
-```{r 10-toy-example-plot, echo = FALSE, warning = FALSE, fig.height = 4, fig.width = 4.35, fig.cap = "Subset data of penguin scaled bill length versus flipper length."}
-base <- ggplot(data, aes(x = flipper_length_scaled, y = bill_length_scaled)) +
+Next, we can create a scatter plot using this data set to see if we can use
+data visualization to detect subtypes or groups in our data set.
+
+```{r 10-toy-example-plot, warning = FALSE, fig.height = 4, fig.width = 4.35, fig.cap = "Subset data of penguin standardized bill length versus standardized flipper length."}
+ggplot(data, aes(x = flipper_length_standardized, y = bill_length_standardized)) +
   geom_point() +
-  xlab("Flipper Length (scaled)") +
-  ylab("Bill Length (scaled)")
-base
+  xlab("Flipper Length (standardized)") +
+  ylab("Bill Length (standardized)")
 ```
 
 Based \index{ggplot}\index{ggplot!geom\_point} on the visualization in Figure \@ref(fig:10-toy-example-plot), we might
-suspect there are a few subtypes of penguins, selected from combinations of
-high/low flipper length and high/low bill length. How do we find this grouping
-automatically, and how do we pick the number of subtypes? The way to rigorously
-separate the data into groups is to use a clustering algorithm.
-In this chapter, we will focus on the *K-means* algorithm, \index{K-means} a widely-used
-and often very effective clustering method, combined with the *elbow method* \index{elbow method} for
-selecting the number of clusters. This procedure will separate the data into 
-the following groups denoted by colour:
-
-```{r 10-toy-example-clustering, echo = FALSE, warning = FALSE, fig.height = 4, fig.width = 5, fig.cap = "Subset data of penguin scaled bill length versus flipper length with coloured groups"}
-ggplot(data, aes(y = bill_length_scaled, x = flipper_length_scaled, colour = cluster)) +
+suspect there are a few subtypes of penguins 
+within the observations we have in our data set.
+We can see ~ 3 groups of observations in Figure \@ref(fig:10-toy-example-plot),
+including:
+
+1. a small flipper and bill length group,
+2. a small flipper length, but large bill length group, and
+3. a large  flipper and bill length group.
+
+This is interesting that we can use data visualization to uncover such patterns,
+but also leads to the questions of how might we find 
+these kind of grouping patterns automatically, 
```

Comment:
this sentence blends a statement and a question mark in an odd way... I think it's just a statement

Reply:
fixed